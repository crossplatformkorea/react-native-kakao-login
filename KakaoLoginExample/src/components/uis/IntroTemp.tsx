import React from 'react';
import {ScrollView} from 'react-native';
import styled from 'styled-components/native';

const ContentWrapper = styled.View`
  flex-direction: column;
  height: 100%;
  width: 100%;
  padding: 24px;
  justify-content: flex-start;
  align-items: center;
`;

const StyledText = styled.Text`
  font-size: 18px;
  line-height: 27px;
  color: ${({theme}) => theme.text};
`;

type Props = {
  result: string;
};

function IntroView({result}: Props): React.ReactElement {
  return (
    <ContentWrapper>
      <ScrollView>
        <StyledText>{result}</StyledText>
      </ScrollView>
    </ContentWrapper>
  );
}

export default IntroView;
