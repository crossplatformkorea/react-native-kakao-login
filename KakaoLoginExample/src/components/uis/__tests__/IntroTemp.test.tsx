import 'react-native';

import React, {ReactElement} from 'react';
import {RenderAPI, cleanup, render} from '@testing-library/react-native';
import {createTestElement, createTestProps} from '../../../../test/testUtils';

import IntroTemp from '../IntroTemp';

let props: any;
let component: ReactElement;
let testingLib: RenderAPI;

describe('[IntroTemp] render', () => {
  props = createTestProps();

  component = createTestElement(<IntroTemp {...props} />);

  afterEach(cleanup);

  it('renders without crashing', () => {
    testingLib = render(component);

    const baseElement = testingLib.toJSON();

    expect(baseElement).toMatchSnapshot();
    expect(baseElement).toBeTruthy();
  });
});
