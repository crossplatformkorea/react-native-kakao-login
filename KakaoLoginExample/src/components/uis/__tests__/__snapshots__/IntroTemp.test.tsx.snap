// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[IntroTemp] render renders without crashing 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "column",
        "height": "100%",
        "justifyContent": "flex-start",
        "width": "100%",
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#000000",
          "fontSize": 18,
          "lineHeight": 27,
        },
        Object {
          "marginTop": 100,
        },
      ]
    }
  >
    
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#000000",
          "fontSize": 18,
          "lineHeight": 27,
        },
      ]
    }
  >
    
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#000000",
          "fontSize": 18,
          "lineHeight": 27,
        },
      ]
    }
  >
    
  </Text>
</View>
`;
